core/container.go

```diff
@@ -1699,6 +1695,25 @@ func (container *Container) command(opts ContainerExecOpts) ([]string, error) {
 	return args, nil
 }
 
+// need bikeshedding for the name. this is basicaly SecretArg + actual secret
+type SecretArgInternal struct {
+	Name     string
+	SecretID SecretID
+	Secret   *Secret
```

Comment:
I think we can remove the need for this by having a `dagql.Instance` here. e.g. https://github.com/dagger/dagger/blob/e40bb3b8f341ffe6ad70ad46d6912716cdf711c7/core/service.go#L50-L51